action.c

```diff
@@ -1257,11 +1277,22 @@ actionCommit(action_t *__restrict__ const pThis, wti_t *__restrict__ const pWti)
 	bDone = 0;
 	do {
 		iRet = actionTryCommit(pThis, pWti);
-		DBGPRINTF("actionCommit, in retry loop, iRet %d\n", iRet);
+		DBGPRINTF("actionCommit, action %d, in retry loop, iRet %d\n",
+			pThis->iActionNbr, iRet);
+		if(iRet == RS_RET_FORCE_TERM) {
```

Comment:
no, because RS_RET_FORCE_TERM really tells us to finish as quickly as possible. So it is better to check before each potentially time-consuming task. But I appreciate your note!

side-note: while this code is better than what was there previously, some more refactoring is strongly desired and on my todo list. I just want to push this here as quickly as possible, as it definitely solves a number of hard-to-track issues (that we have seen on the issue trackers and mailing list).